mlflow/data/spark_dataset.py

```diff
@@ -106,12 +106,39 @@ def profile(self) -> Optional[Any]:
         """
         A profile of the dataset. May be None if no profile is available.
         """
-        # use Spark RDD countApprox to get approximate count since count() may be expensive
-        approx_count = self.df.rdd.countApprox(timeout=1000, confidence=0.90)
```

Comment:
PySpark countApprox doesn't respect the timeout. Scala does.

Reply:
Interesting find. How do we know this? The docs indicate the timeout works...
https://spark.apache.org/docs/3.1.3/api/python/reference/api/pyspark.RDD.countApprox.html

Reply:
The docs...lie :/. I tried this out and a notebook and found it was taking 10+ seconds to get a count on a few thousand record example DF.

I then read https://stackoverflow.com/questions/51238557/rdd-countapproximate-taking-far-longer-than-requested-timeout and determined that the problem is PySpark uses a `.finalValue()` call which blocks until Spark finishes a full count irrespective of timeout (confirmed in PySpark source: https://spark.apache.org/docs/3.1.3/api/python/_modules/pyspark/rdd.html#RDD.countApprox).

We need to use `initialValue()` instead, as indicated in https://stackoverflow.com/questions/51238557/rdd-countapproximate-taking-far-longer-than-requested-timeout, which requires Scala Spark.